src/main/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java

```diff
@@ -187,109 +182,13 @@ public ItemStack getResult()
-                                                getMinLevel(),
-                                                getMaxLevel()));
-
-        if (!toolTypeResult)
-        {
-            return (stack.getItem() instanceof ShieldItem && toolClass.equals(ToolType.SHIELD))
-                     || (stack.getItem() instanceof FlintAndSteelItem && toolClass.equals(ToolType.FLINT_N_STEEL));
-        }
-
-        return toolTypeResult;
-    }
-
-    private Set<String> getToolClasses(final ItemStack stack)
-    {
-        final Set<String> set = new HashSet<>();
-
-        if (ItemStackUtils.isEmpty(stack))
-        {
-            return set;
-        }
-
-        if (stack.canPerformAction(ItemAbilities.AXE_DIG))
-        {
-            set.add(ToolType.AXE.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.PICKAXE_DIG))
-        {
-            set.add(ToolType.PICKAXE.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.SHOVEL_DIG))
-        {
-            set.add(ToolType.SHOVEL.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.HOE_DIG))
-        {
-            set.add(ToolType.HOE.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.SWORD_SWEEP))
-        {
-            set.add(ToolType.SWORD.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.SHEARS_DIG))
-        {
-            set.add(ToolType.SHEARS.getName());
-        }
-
-        if (stack.canPerformAction(ItemAbilities.FISHING_ROD_CAST))
-        {
-            set.add(ToolType.FISHINGROD.getName());
-        }
-
-        if (stack.getItem() instanceof BowItem)
-        {
-            set.add("bow");
-        }
-        else if (stack.getItem() instanceof SwordItem || Compatibility.isTinkersWeapon(stack))
-        {
-            set.add("weapon");
-        }
-        else if (stack.getItem() instanceof ShieldItem)
-        {
-            set.add("shield");
-        }
-        else if (stack.getItem() instanceof ArmorItem)
-        {
-            /*
-             * There is no armor class for each type of armor.
-             * So what we need to do is check the equipment Slot of this
-             * armor to send back what type of armor this if for the request
-             * system.
-             */
-            final ArmorItem armor = (ArmorItem) stack.getItem();
-            if (armor.getEquipmentSlot() == EquipmentSlot.CHEST)
-            {
-                set.add("chestplate");
-            }
-            else if (armor.getEquipmentSlot() == EquipmentSlot.FEET)
-            {
-                set.add("boots");
-            }
-            else if (armor.getEquipmentSlot() == EquipmentSlot.HEAD)
-            {
-                set.add("helmet");
-            }
-            else if (armor.getEquipmentSlot() == EquipmentSlot.LEGS)
-            {
-                set.add("leggings");
-            }
-        }
-        return set;
+        IToolType toolType = getToolClass();
```

Comment:
@Raycoms wasn't the new tool system already ported to 1.21?

Reply:
should be, ye

Reply:
Oops I was way behind, let me rebase and use the new tool stuff.

Reply:
Updated! This no longer touches the tools code, the new stuff you have looks good already.